CMakeLists.txt

```diff
@@ -111,9 +111,13 @@ endif()
 # FORCE is required for the "asked for but missing tool" case
 set(BUILD_RUST ${BUILD_RUST} CACHE BOOL "Build and run Rust code" FORCE)
 
+if (EMSCRIPTEN)
+  set(HARDWARE_IS_SCARY TRUE CACHE BOOL "Do not build code that touches hardware" FORCE)
```

Comment:
This is a wrong way to use `set(CACHE)`. Why do you need a user-visible switch here anyway? How is 
```
set(BUILD_HARDWARE_TOOLS TRUE)
if(EMSCRIPTEN)
  set(BUILD_HARDWARE_TOOLS TRUE)
endif()
```
not enough?

Reply:
The idea was that I did want this to be a user-visible feature switch. The CACHE and FORCE is so that the cmake script can forcefully override the user's choice if the user's choice isn't feasible. This is also done for BUILD_DOC and BUILD_RUST. Is this wrong?

Reply:
Yes. The variable is not defined at all for non-emscripten builds (it should be defined such that tools are built), which means it won't show up in lists of configuration variables, GUIs, etc.

There is no point in defining a cache variable that is always of the same value. Define a normal one, it will override any cache variable even if it exists. Bonus points for printing a message if the variable was defined before, in any way.